web-app/src/views/views.py

```diff
@@ -0,0 +1,37 @@
+from celery.result import AsyncResult
+from flask import Blueprint
+from flask import request, jsonify
+
+from tasks import tasks
+
+import os
+import tempfile
+
+bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+@bp.get("/result/<id>")
+def result(id: str) -> dict[str, object]:
+    result = AsyncResult(id)
+    ready = result.ready()
+    return {
+        "ready": ready,
+        "successful": result.successful() if ready else None,
```

Comment:
Suggestion: I'd put the whole Celery [task state](https://docs.celeryq.dev/en/stable/reference/celery.states.html#std-state-READY_STATES) description instead of just 'successful' flag, e.g.:
`"state": result.state`
so that we have more visibility on task progress directly from browser's dev tools (will be very helpful when we run in the cloud).
Also, we may want to mark task failure in the UI.